src/overlays/actors/ovl_Bg_Ice_Shelter/z_bg_ice_shelter.c

```diff
@@ -71,34 +80,42 @@ static ColliderCylinderInit D_80891738 = {
     { 0, 0, 0, { 0, 0, 0 } },
 };
 
-void func_80890740(BgIceShelter* this, PlayState* play) {
+/**
+ * Initializes either one or both cylinder colliders, depending on the actor's type.
+ */
+void BgIceShelter_InitColliders(BgIceShelter* this, PlayState* play) {
     static s16 cylinderRadii[] = { 47, 33, 44, 41, 100 };
     static s16 cylinderHeights[] = { 80, 54, 90, 60, 200 };
     s32 pad;
-    s32 type = (this->dyna.actor.params >> 8) & 7;
+    s32 type = GET_TYPE(this);
 
+    // All types use at least one collider in order to detect blue fire
     Collider_InitCylinder(play, &this->cylinder1);
-    Collider_SetCylinder(play, &this->cylinder1, &this->dyna.actor, &D_8089170C);
+    Collider_SetCylinder(play, &this->cylinder1, &this->dyna.actor, &sCylinderInit1);
     Collider_UpdateCylinder(&this->dyna.actor, &this->cylinder1);
 
     this->cylinder1.dim.radius = cylinderRadii[type];
     this->cylinder1.dim.height = cylinderHeights[type];
 
-    if (type == 0 || type == 1 || type == 4) {
+    // The wall and platform types use DynaPoly for collision, so they don't need the second collider
+    if (type == RED_ICE_LARGE || type == RED_ICE_SMALL || type == RED_ICE_KING_ZORA) {
         Collider_InitCylinder(play, &this->cylinder2);
-        Collider_SetCylinder(play, &this->cylinder2, &this->dyna.actor, &D_80891738);
+        Collider_SetCylinder(play, &this->cylinder2, &this->dyna.actor, &sCylinderInit2);
         Collider_UpdateCylinder(&this->dyna.actor, &this->cylinder2);
         this->cylinder2.dim.radius = cylinderRadii[type];
         this->cylinder2.dim.height = cylinderHeights[type];
     }
 
-    if (type == 4) {
+    if (type == RED_ICE_KING_ZORA) {
         this->cylinder1.dim.pos.z += 30;
         this->cylinder2.dim.pos.z += 30;
     }
 }
 
-void func_80890874(BgIceShelter* this, PlayState* play, CollisionHeader* collision, s32 moveFlag) {
+/**
+ * Initializes the DynaPoly for the types that use it.
+ */
+void BgIceShelter_InitDynaPoly(BgIceShelter* this, PlayState* play, CollisionHeader* collision, s32 moveFlag) {
```

Comment:
You can look for `InitDynaPoly` in the codebase to see how much this function has been copypasted

Since it's almost always the same thing I think we can skip comments on it?

Reply:
Done